src/components/scene/canvas.js

```diff
@@ -29,6 +29,8 @@ module.exports.Component = register('canvas', {
       canvas.style.height = data.height + '%';
       canvas.style.width = data.width + '%';
       scene.appendChild(canvas);
+    } else {
+      canvas.dataset.aframeProvided = true;
```

Comment:
i didn't fully understand the architecture. is everything that happens in this component the result of a provided canvas?